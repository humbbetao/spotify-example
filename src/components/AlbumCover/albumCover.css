.album-item {
  width: 200px;
  height: 300px;
  display: block;
  display: flex;
  flex-direction: column;
  text-align: center;
  justify-content: center;
  align-items: center;
  margin: 16px;
}
.album-cover {
  width: 80%;
  height: 80%;
  min-width: 200px;
  min-height: 200px;
  max-width: 200px;
  max-height: 200px;
  object-fit: contain;
  box-sizing: border-box;
  background-color: var(--third-color);
  background-color: var(--primary-color);
}

.title-album {
  color: var(--third-color);
  font-family: Roboto;
  text-align: center;
  padding: 12px 4px;
  font-size: 14px;
  white-space: pre-wrap;
}
.artist-album {
  color: var(--primary-color);
  font-family: Roboto;
  text-align: center;
  padding: 12px 4px;
  font-size: 14px;
  white-space: pre-wrap;
}
