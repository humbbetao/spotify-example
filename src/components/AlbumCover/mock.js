import { mount } from 'enzyme'
import React from 'react'
import { createMockStore } from 'redux-test-utils'
import { Provider } from 'react-redux'
import { Router, BrowserRouter } from 'react-router-dom'
import { createHistory } from 'react-router-dom'
import PropTypes from 'prop-types'

export const mountWithRouterAndStore = (Component, storeState = {}) => {
  const store = createMockStore(storeState)

  return mount(
    <Provider store={store}>
      <BrowserRouter>{Component}</BrowserRouter>
    </Provider>
  )
}

export function mountWithRouterAndStore2(Component, storeState = {}) {
  const store = createMockStore(storeState)
  // const history = createHistory()
  const history = {}
  const options = {
    context: { store, router: { history, route: { location: {} } } },
    childContextTypes: {
      store: PropTypes.object.isRequired,
      router: PropTypes.object.isRequired,
    },
  }

  return mount(Component, options)
}
