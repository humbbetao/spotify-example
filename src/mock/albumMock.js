const arrayMock = [
  {
    album_type: 'single',
    artists: [
      {
        external_urls: {
          spotify: 'https://open.spotify.com/artist/6rFkZxhheU9l1nODgZm4SP',
        },
        href: 'https://api.spotify.com/v1/artists/6rFkZxhheU9l1nODgZm4SP',
        id: '6rFkZxhheU9l1nODgZm4SP',
        name: 'Junior Lord',
        type: 'artist',
        uri: 'spotify:artist:6rFkZxhheU9l1nODgZm4SP',
      },
    ],
    available_markets: [
      'AD',
      'AE',
      'AR',
      'AT',
      'AU',
      'BE',
      'BG',
      'BH',
      'BO',
      'BR',
      'CA',
      'CH',
      'CL',
      'CO',
      'CR',
      'CY',
      'CZ',
      'DE',
      'DK',
      'DO',
      'DZ',
      'EC',
      'EE',
      'EG',
      'ES',
      'FI',
      'FR',
      'GB',
      'GR',
      'GT',
      'HK',
      'HN',
      'HU',
      'ID',
      'IE',
      'IL',
      'IN',
      'IS',
      'IT',
      'JO',
      'JP',
      'KW',
      'LB',
      'LI',
      'LT',
      'LU',
      'LV',
      'MA',
      'MC',
      'MT',
      'MX',
      'MY',
      'NI',
      'NL',
      'NO',
      'NZ',
      'OM',
      'PA',
      'PE',
      'PH',
      'PL',
      'PS',
      'PT',
      'PY',
      'QA',
      'RO',
      'SA',
      'SE',
      'SG',
      'SK',
      'SV',
      'TH',
      'TN',
      'TR',
      'TW',
      'US',
      'UY',
      'VN',
      'ZA',
    ],
    external_urls: {
      spotify: 'https://open.spotify.com/album/2Yv3YuHlHmeSnMYdF5hhzS',
    },
    href: 'https://api.spotify.com/v1/albums/2Yv3YuHlHmeSnMYdF5hhzS',
    id: '2Yv3YuHlHmeSnMYdF5hhzS',
    images: [
      {
        height: 640,
        url: 'https://i.scdn.co/image/ab67616d0000b273c2744f6a4677f1677b3eca12',
        width: 640,
      },
      {
        height: 300,
        url: 'https://i.scdn.co/image/ab67616d00001e02c2744f6a4677f1677b3eca12',
        width: 300,
      },
      {
        height: 64,
        url: 'https://i.scdn.co/image/ab67616d00004851c2744f6a4677f1677b3eca12',
        width: 64,
      },
    ],
    name: 'Rihanna',
    release_date: '2019-06-21',
    release_date_precision: 'day',
    total_tracks: 1,
    type: 'album',
    uri: 'spotify:album:2Yv3YuHlHmeSnMYdF5hhzS',
  },
  {
    album_type: 'album',
    artists: [
      {
        external_urls: {
          spotify: 'https://open.spotify.com/artist/5pKCCKE2ajJHZ9KAiaK11H',
        },
        href: 'https://api.spotify.com/v1/artists/5pKCCKE2ajJHZ9KAiaK11H',
        id: '5pKCCKE2ajJHZ9KAiaK11H',
        name: 'Rihanna',
        type: 'artist',
        uri: 'spotify:artist:5pKCCKE2ajJHZ9KAiaK11H',
      },
    ],
    available_markets: [
      'AD',
      'AE',
      'AR',
      'AT',
      'AU',
      'BE',
      'BG',
      'BH',
      'BO',
      'BR',
      'CA',
      'CH',
      'CL',
      'CO',
      'CR',
      'CY',
      'CZ',
      'DE',
      'DK',
      'DO',
      'DZ',
      'EC',
      'EE',
      'EG',
      'ES',
      'FI',
      'FR',
      'GB',
      'GR',
      'GT',
      'HK',
      'HN',
      'HU',
      'ID',
      'IE',
      'IL',
      'IN',
      'IS',
      'IT',
      'JO',
      'JP',
      'KW',
      'LB',
      'LI',
      'LT',
      'LU',
      'LV',
      'MA',
      'MC',
      'MT',
      'MX',
      'MY',
      'NI',
      'NL',
      'NO',
      'NZ',
      'OM',
      'PA',
      'PE',
      'PH',
      'PL',
      'PS',
      'PT',
      'PY',
      'QA',
      'RO',
      'SA',
      'SE',
      'SG',
      'SK',
      'SV',
      'TH',
      'TN',
      'TR',
      'TW',
      'US',
      'UY',
      'VN',
      'ZA',
    ],
    external_urls: {
      spotify: 'https://open.spotify.com/album/4UlGauD7ROb3YbVOFMgW5u',
    },
    href: 'https://api.spotify.com/v1/albums/4UlGauD7ROb3YbVOFMgW5u',
    id: '4UlGauD7ROb3YbVOFMgW5u',
    images: [
      {
        height: 640,
        url: 'https://i.scdn.co/image/ab67616d0000b27333c6b920eabcf4c00d7a1093',
        width: 640,
      },
      {
        height: 300,
        url: 'https://i.scdn.co/image/ab67616d00001e0233c6b920eabcf4c00d7a1093',
        width: 300,
      },
      {
        height: 64,
        url: 'https://i.scdn.co/image/ab67616d0000485133c6b920eabcf4c00d7a1093',
        width: 64,
      },
    ],
    name: 'ANTI (Deluxe)',
    release_date: '2016-01-28',
    release_date_precision: 'day',
    total_tracks: 16,
    type: 'album',
    uri: 'spotify:album:4UlGauD7ROb3YbVOFMgW5u',
  },
  {
    album_type: 'album',
    artists: [
      {
        external_urls: {
          spotify: 'https://open.spotify.com/artist/5pKCCKE2ajJHZ9KAiaK11H',
        },
        href: 'https://api.spotify.com/v1/artists/5pKCCKE2ajJHZ9KAiaK11H',
        id: '5pKCCKE2ajJHZ9KAiaK11H',
        name: 'Rihanna',
        type: 'artist',
        uri: 'spotify:artist:5pKCCKE2ajJHZ9KAiaK11H',
      },
    ],
    available_markets: [
      'AD',
      'AE',
      'AR',
      'AT',
      'AU',
      'BE',
      'BG',
      'BH',
      'BO',
      'BR',
      'CA',
      'CH',
      'CL',
      'CO',
      'CR',
      'CY',
      'CZ',
      'DE',
      'DK',
      'DO',
      'DZ',
      'EC',
      'EE',
      'EG',
      'ES',
      'FI',
      'FR',
      'GB',
      'GR',
      'GT',
      'HK',
      'HN',
      'HU',
      'ID',
      'IE',
      'IL',
      'IN',
      'IS',
      'IT',
      'JO',
      'JP',
      'KW',
      'LB',
      'LI',
      'LT',
      'LU',
      'LV',
      'MA',
      'MC',
      'MT',
      'MX',
      'MY',
      'NI',
      'NL',
      'NO',
      'NZ',
      'OM',
      'PA',
      'PE',
      'PH',
      'PL',
      'PS',
      'PT',
      'PY',
      'QA',
      'RO',
      'SA',
      'SE',
      'SG',
      'SK',
      'SV',
      'TH',
      'TN',
      'TR',
      'TW',
      'US',
      'UY',
      'VN',
      'ZA',
    ],
    external_urls: {
      spotify: 'https://open.spotify.com/album/3JSWZWeTHF4HDGt5Eozdy7',
    },
    href: 'https://api.spotify.com/v1/albums/3JSWZWeTHF4HDGt5Eozdy7',
    id: '3JSWZWeTHF4HDGt5Eozdy7',
    images: [
      {
        height: 640,
        url: 'https://i.scdn.co/image/ab67616d0000b273f9f27162ab1ed45b8d7a7e98',
        width: 640,
      },
      {
        height: 300,
        url: 'https://i.scdn.co/image/ab67616d00001e02f9f27162ab1ed45b8d7a7e98',
        width: 300,
      },
      {
        height: 64,
        url: 'https://i.scdn.co/image/ab67616d00004851f9f27162ab1ed45b8d7a7e98',
        width: 64,
      },
    ],
    name: 'Good Girl Gone Bad: Reloaded',
    release_date: '2008-06-02',
    release_date_precision: 'day',
    total_tracks: 15,
    type: 'album',
    uri: 'spotify:album:3JSWZWeTHF4HDGt5Eozdy7',
  },
  {
    album_type: 'album',
    artists: [
      {
        external_urls: {
          spotify: 'https://open.spotify.com/artist/5pKCCKE2ajJHZ9KAiaK11H',
        },
        href: 'https://api.spotify.com/v1/artists/5pKCCKE2ajJHZ9KAiaK11H',
        id: '5pKCCKE2ajJHZ9KAiaK11H',
        name: 'Rihanna',
        type: 'artist',
        uri: 'spotify:artist:5pKCCKE2ajJHZ9KAiaK11H',
      },
    ],
    available_markets: [
      'AD',
      'AE',
      'AR',
      'AT',
      'AU',
      'BE',
      'BG',
      'BH',
      'BO',
      'BR',
      'CH',
      'CL',
      'CO',
      'CR',
      'CY',
      'CZ',
      'DE',
      'DK',
      'DO',
      'DZ',
      'EC',
      'EE',
      'EG',
      'ES',
      'FI',
      'FR',
      'GB',
      'GR',
      'GT',
      'HK',
      'HN',
      'HU',
      'ID',
      'IL',
      'IN',
      'IS',
      'IT',
      'JO',
      'JP',
      'KW',
      'LB',
      'LI',
      'LT',
      'LU',
      'LV',
      'MA',
      'MC',
      'MT',
      'MY',
      'NI',
      'NL',
      'NO',
      'NZ',
      'OM',
      'PA',
      'PE',
      'PH',
      'PL',
      'PS',
      'PT',
      'PY',
      'QA',
      'RO',
      'SA',
      'SE',
      'SG',
      'SK',
      'SV',
      'TH',
      'TN',
      'TR',
      'TW',
      'UY',
      'VN',
      'ZA',
    ],
    external_urls: {
      spotify: 'https://open.spotify.com/album/7vN82vd1Vq44fjlhjfvHJp',
    },
    href: 'https://api.spotify.com/v1/albums/7vN82vd1Vq44fjlhjfvHJp',
    id: '7vN82vd1Vq44fjlhjfvHJp',
    images: [
      {
        height: 640,
        url: 'https://i.scdn.co/image/ab67616d0000b273be99e0ef9d3e6a1b55d94e78',
        width: 640,
      },
      {
        height: 300,
        url: 'https://i.scdn.co/image/ab67616d00001e02be99e0ef9d3e6a1b55d94e78',
        width: 300,
      },
      {
        height: 64,
        url: 'https://i.scdn.co/image/ab67616d00004851be99e0ef9d3e6a1b55d94e78',
        width: 64,
      },
    ],
    name: 'Loud',
    release_date: '2010-11-16',
    release_date_precision: 'day',
    total_tracks: 12,
    type: 'album',
    uri: 'spotify:album:7vN82vd1Vq44fjlhjfvHJp',
  },
]

export default arrayMock
