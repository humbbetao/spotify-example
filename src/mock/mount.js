import { mount } from 'enzyme'
import React from 'react'
import { createMockStore } from 'redux-test-utils'
import { Provider } from 'react-redux'
import { Router, BrowserRouter, Switch } from 'react-router-dom'
import PropTypes from 'prop-types'

export const mountWithRouterAndStore = (Component, storeState = {}) => {
  const store = createMockStore(storeState)

  return mount(
    <Provider store={store}>
      <BrowserRouter>{Component}</BrowserRouter>
    </Provider>
  )
}

export const mountWithRouterAndStoreWithRoutes = (
  Component,
  storeState = {},
  path
) => {
  const store = createMockStore(storeState)

  return mount(
    <Provider store={store}>
      <BrowserRouter>
        <Router path={path} Component={Component}></Router>
      </BrowserRouter>
    </Provider>
  )
}

export function mountWithRouterAndStore2(Component, storeState = {}) {
  const store = createMockStore(storeState)
  const history = {}
  const options = {
    context: { store, router: { history, route: { location: {} } } },
    childContextTypes: {
      store: PropTypes.object.isRequired,
      router: PropTypes.object.isRequired,
    },
  }

  return mount(Component, options)
}
