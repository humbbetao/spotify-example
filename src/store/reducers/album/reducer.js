import ActionTypes from './actionTypes'

const INITIAL_STATE = {
  query: '',
  hasAlbumsRecentlySearched: false,
  albums: [],
}
export default function reducer(state = INITIAL_STATE, action) {
  switch (action.type) {
    case ActionTypes.SEARCH_ALBUMS:
      return {
        ...state,
        query: action.payload.query,
      }
    case ActionTypes.SEARCH_ALBUMS_SUCCESS:
      debugger
      return {
        ...state,
        albums: action.payload.albums,
        hasAlbumsRecentlySearched: true,
      }

    case ActionTypes.SEARCH_ALBUMS_ERROR:
      return {
        ...state,
        albunsError: action.payload.albunsError,
      }

    default:
      return state
  }
}
