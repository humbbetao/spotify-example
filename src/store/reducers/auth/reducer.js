import ActionTypes from './actionTypes'
export default function reducer(state = {}, action) {
  switch (action.type) {
    case ActionTypes.GET_TOKEN_SUCCESS:
      return {
        ...state,
        token: action.payload.token,
      }

    default:
      return state
  }
}
