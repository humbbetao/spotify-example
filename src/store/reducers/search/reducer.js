import ActionTypes from './actionTypes'

export default function browserReducer(state = {}, action) {
  switch (action.type) {
    case ActionTypes.SEARCH_ALBUNS:
      return {
        ...state,
        query: action.payload.query,
      }
    case ActionTypes.SEARCH_ALBUNS_SUCCESS:
      return {
        ...state,
        albuns: action.payload.albuns,
      }

    case ActionTypes.SEARCH_ALBUNS_ERROR:
      return {
        ...state,
        albunsError: action.payload.albunsError,
      }

    default:
      return state
  }
}
