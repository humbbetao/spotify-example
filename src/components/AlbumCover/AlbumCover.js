import React from 'react'
import P from 'components/P'
import './albumCover.css'
import { useHistory } from 'react-router-dom'
import { useDispatch } from 'react-redux'
import SongActions from 'store/reducers/songs/actionCreators'

export default function AlbumCover({ album }) {
  const history = useHistory()
  const dispatch = useDispatch()
  if (!album) return null
  const artist = album.artists[0].name

  const handleOnClickOnSong = () => {
    dispatch(SongActions.setAlbum(album))
    history.push(`/albums/${artist}/${album.id}`)
  }

  return (
    <li className="album-item" onClick={handleOnClickOnSong}>
      <picture>
        {album.images &&
          album.images.map(image => (
            <source
              key={image.url}
              media={`(max-width: ${image.width * 2.5}px)`}
              srcSet={image.url}
              alt={album.name}
              className="album-cover"
            />
          ))}
        <img
          className="album-cover"
          src={album.images[0].url}
          alt={album.name}
        />
      </picture>

      <P className="title-album">{artist}</P>
      <P className="artist-album">{album.name}</P>
    </li>
  )
}
