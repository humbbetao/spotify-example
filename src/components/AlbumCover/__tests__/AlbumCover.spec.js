import React from 'react'
import AlbumCover from '../AlbumCover'
import { mountWithRouterAndStore } from 'mock/mount'
import { INITIAL_STATE } from 'store/reducers/songs/reducer'
import albumMock from 'mock/albumMock'

describe('MyComponent', () => {
  it('should render correctly in when album is empty', () => {
    const reduxState = { state: { songs: { INITIAL_STATE } } }
    console.log(INITIAL_STATE)
    console.log(reduxState)
    const wrapper = mountWithRouterAndStore(<AlbumCover />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render correctly in when album is no empty', () => {
    const reduxState = { state: { songs: { INITIAL_STATE } } }
    console.log(INITIAL_STATE)
    console.log(reduxState)
    const album = albumMock[0]
    const wrapper = mountWithRouterAndStore(<AlbumCover album={album} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render correctly in when album is no empty', () => {
    const reduxState = { state: { songs: { INITIAL_STATE } } }
    console.log(INITIAL_STATE)
    console.log(reduxState)
    const album = albumMock[0]
    const wrapper = mountWithRouterAndStore(<AlbumCover album={album} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should call onClick when is clicked in card', () => {
    const reduxState = { state: { songs: { INITIAL_STATE } } }
    console.log(INITIAL_STATE)
    console.log(reduxState)
    const album = albumMock[0]
    const wrapper = mountWithRouterAndStore(<AlbumCover album={album} />, {})

    expect(wrapper.find('.album-item').exists()).toBeTruthy()
    expect(wrapper.exists('.album-item')).toBeTruthy()
    wrapper.find('.album-item').simulate('click')
    console.log(wrapper.find('.album-item'))
    expect(wrapper.exists('.album-item')).toBeFalsy()
  })
})
