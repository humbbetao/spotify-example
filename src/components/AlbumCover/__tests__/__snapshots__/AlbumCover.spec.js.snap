// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should call onClick when is clicked in card 1`] = `ReactWrapper {}`;

exports[`MyComponent should render correctly in when album is empty 1`] = `ReactWrapper {}`;

exports[`MyComponent should render correctly in when album is no empty 1`] = `ReactWrapper {}`;

exports[`MyComponent should render correctly in when album is no empty 2`] = `ReactWrapper {}`;
